src/Rules/PackageData.hs

```diff
@@ -3,115 +3,32 @@ module Rules.PackageData (buildPackageData) where
     dir -/- "inplace-pkg-config" %> \conf -> do
-        path     <- buildPath context
-        dataFile <- pkgDataFile context
-        need [dataFile] -- ghc-cabal builds inplace package configuration file
-        if package == rts
-        then do
-            genPath <- buildRoot <&> (-/- generatedDir)
-            rtsPath <- rtsBuildPath
-            need [rtsConfIn]
-            build $ target context HsCpp [rtsConfIn] [conf]
-            fixFile conf $ unlines
-                         . map
-                         ( replace "\"\"" ""
-                         . replace "rts/dist/build" rtsPath
-                         . replace "includes/dist-derivedconstants/header" genPath )
-                         . lines
-        else
-            fixFile conf $ unlines . map (replace (path </> "build") path) . lines
-
-    priority 2.0 $ when (nonCabalContext context) $ dir -/- "package-data.mk" %>
-        generatePackageData context
-
-generatePackageData :: Context -> FilePath -> Action ()
-generatePackageData context@Context {..} file = do
-    orderOnly =<< interpretInContext context generatedDependencies
-    asmSrcs <- packageAsmSources package
-    cSrcs   <- packageCSources   package
-    cmmSrcs <- packageCmmSources package
-    genPath <- buildRoot <&> (-/- generatedDir)
-    writeFileChanged file . unlines $
-        [ "S_SRCS = "   ++ unwords asmSrcs                                  ] ++
-        [ "C_SRCS = "   ++ unwords cSrcs                                    ] ++
-        [ "CMM_SRCS = " ++ unwords cmmSrcs                                  ] ++
-        [ "DEP_EXTRA_LIBS = m"                 | package == hp2ps           ] ++
-        [ "CC_OPTS = -I" ++ genPath            | package `elem` [hp2ps, rts]] ++
-        [ "MODULES = Main"                     | package == ghcCabal        ] ++
-        [ "HS_SRC_DIRS = ."                    | package == ghcCabal        ]
-    putSuccess $ "| Successfully generated " ++ file
-
-packageCSources :: Package -> Action [FilePath]
-packageCSources pkg
-    | pkg /= rts = getDirectoryFiles (pkgPath pkg) ["*.c"]
-    | otherwise  = do
-        windows <- windowsHost
-        sources <- fmap (map unifyPath) . getDirectoryFiles (pkgPath pkg) .
-            map (-/- "*.c") $ [ ".", "hooks", "sm", "eventlog", "linker" ] ++
-                              [ if windows then "win32" else "posix"     ]
-        return sources
-
-packageAsmSources :: Package -> Action [FilePath]
-packageAsmSources pkg
-    | pkg /= rts = return []
-    | otherwise  = do
-        buildAdjustor   <- anyTargetArch ["i386", "powerpc", "powerpc64"]
-        buildStgCRunAsm <- anyTargetArch ["powerpc64le"]
-        return $ [ "AdjustorAsm.S" | buildAdjustor   ]
-              ++ [ "StgCRunAsm.S"  | buildStgCRunAsm ]
-
-packageCmmSources :: Package -> Action [FilePath]
-packageCmmSources pkg
-    | pkg /= rts = return []
-    | otherwise  = do
+      dataFile <- pkgDataFile context
```

Comment:
Wait, why do we still need `pkgDataFile`?

Reply:
Well, there's no rule to generate it anymore anyways, I'm getting rid of it in an upcoming commit.